crates/sui-indexer/src/apis/read_api.rs

```diff
@@ -464,6 +468,62 @@ where
             .get_checkpoints(cursor, limit, descending_order)
             .await;
     }
+
+    async fn query_objects(
+        &self,
+        query: SuiObjectResponseQuery,
+        cursor: Option<ObjectID>,
+        limit: Option<usize>,
+        descending_order: Option<bool>,
+        at_checkpoint: Option<CheckpointId>,
+    ) -> RpcResult<ObjectsPage> {
+        // Pagination for objects will take in the object and also the checkpoint
+        let limit = validate_limit(limit, QUERY_MAX_RESULT_LIMIT_OBJECTS)?;
+        let is_descending = descending_order.unwrap_or_default();
+        let object_id = match cursor {
+            Some(id) => id,
+            None => ObjectID::ZERO,
+        };
```

Comment:
if is_descending is true, None cursor should be latest object ID rather than 0;
it's easier to handle None in pg_indexer_store